gpii/node_modules/flowManager/src/RepositorySolutionsLoader.js

```diff
@@ -0,0 +1,115 @@
+/*!
+GPII Full SHA Revision DataSource
+
+Copyright 2020 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    path = require("path"),
+    gpii = fluid.registerNamespace("gpii");
+
+require("kettle");
+
+fluid.registerNamespace("gpii.flowManager.repositorySolutionsLoader");
+
+fluid.defaults("gpii.flowManager.repositorySolutionsLoader", {
+    gradeNames: ["kettle.dataSource.URL"],
+    urlTemplate: "%prefix/%revision/%suffix/%fileName",
```

Comment:
In the case such as in tests, if we don't want to fetch SR from the remote, we could potentially override `urlTemplate` to an empty string which will result in `url` being an empty string. The fetch function first checks if `url` is provided, if not, skip fetching from the remote. This gives us an option to skip the remote fetch.